src/GTP.cpp

```diff
@@ -172,6 +176,13 @@ const std::string GTP::s_commands[] = {
     "heatmap",
     "lz-analyze",
     "lz-genmove_analyze",
+    "set_option",
+    ""
+};
+
+const std::string GTP::s_options[] = {
+    "max_tree_size <size in MB> - set the maximum size of a game tree in megabytes",
+    "max_cache_size <size in MB> - set the maximum size of a Neural Network result cache in megabytes",
```

Comment:
No, please. The user specifies a single memory usage maximum and the engine figures out a suitable allocation distribution between both. For example, by keeping the same proportion as whatever default we have now, with some reasonable limits for NNCache (that may already be enforced in that code anyway!).

How is the user supposed to guess what suitable values are here? They shouldn't.

Reply:
Ok. I am ok with making one setting "max_memory" but I would be happy to keep the ratio modifiable if you'll agree. 
For example for people who do a lot of undo/redo while analyzing with Lizzie only the cache size matters and for them even 1:1 would be a good ratio, whereas for basic playing it seems tat 1:10 is roughly fine.
So even if simple end-user would only care about total memory then the tools like Lizzie could request more NNCache heavy ratio.

Reply:
> The user specifies a single memory usage maximum and the engine figures out a suitable allocation distribution between both

@gcp 
Is it acceptable if in the program I count the memory consumption as size of tree + nn cache + network + some empirically achieved magical number?
I don't know the program enough to count what is the whole memory consumption. E.g. when the program starts there seems to much higher memory consumption than the pure network size. 

> How is the user supposed to guess what suitable values are here? They shouldn't.

I am willing to accept your judgment, but I disagree. As I see it there are two classes of users
1) End-user who want to play. 90% of them are using some clients and indeed don't need to know or care about gtp protocol at all.
2) Client authors who are able and potentially willing to understand which is more relevant to the use of their program.